### PR TITLE
ATF-6: Added the feedConfigurationManager

### PR DESCRIPTION
**Pull request by areklalo**
Added functionality:
- load the configuration from the file
- load the configuration from the string
- save the configuration to the file
- JSON validation
- get specific configuration by title

On Monday I will add the unit tests for these methods in this PR.

### CODE REVIEW

File: api/src/main/java/org/openmrs/module/atomfeed/AtomfeedmoduleConstants.java [hunk 1]
```diff
@@ -0,0 +1,8 @@
+package org.openmrs.module.atomfeed;
+
+public final class AtomfeedmoduleConstants {
+	
+	public static final String ATOMFEED_PATH_TO_DAFAULT_CONFIGURATION = "default.json";
```
Comment (pkornowski) [inline api/src/main/java/org/openmrs/module/atomfeed/AtomfeedmoduleConstants.java]: Change filename to defaultAtomfeedConfig

File: api/src/main/java/org/openmrs/module/atomfeed/AtomfeedmoduleConstants.java [hunk 2]
```diff
@@ -0,0 +1,8 @@
...
+	
+	public static final String ATOMFEED_PATH_TO_LOCAL_CONFIGURATION = "local.json";
```
Comment (pkornowski) [inline api/src/main/java/org/openmrs/module/atomfeed/AtomfeedmoduleConstants.java]: localAtomfeedConfig

File: api/src/main/java/org/openmrs/module/atomfeed/api/utils/FeedConfigurationManager.java [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+			loadConfigurations(localConfiguration);
+		}
+	}
+	
+	public FeedConfiguration getConfigurationByTitle(String title) {
+		return feedConfiguration.get(title);
+	}
+	
+	private void loadConfigurations(FeedConfiguration[] feedConfigurations) {
+        HashMap<String, FeedConfiguration> tmp = new HashMap<>();
```
Comment (pkornowski) [inline api/src/main/java/org/openmrs/module/atomfeed/api/utils/FeedConfigurationManager.java]: Fix indentation.

File: api/src/main/java/org/openmrs/module/atomfeed/AtomfeedmoduleConstants.java (hunk 1 above)
Comment (areklalo) [reply api/src/main/java/org/openmrs/module/atomfeed/AtomfeedmoduleConstants.java]: Corrected.

File: api/src/main/java/org/openmrs/module/atomfeed/AtomfeedmoduleConstants.java (hunk 2 above)
Comment (areklalo) [reply api/src/main/java/org/openmrs/module/atomfeed/AtomfeedmoduleConstants.java]: image

Comment (areklalo) [reply api/src/main/java/org/openmrs/module/atomfeed/AtomfeedmoduleConstants.java]: Corrected.

File: api/src/main/java/org/openmrs/module/atomfeed/api/utils/FeedConfigurationManager.java (hunk 3 above)
Comment (areklalo) [reply api/src/main/java/org/openmrs/module/atomfeed/api/utils/FeedConfigurationManager.java]: Fixed.

File: api/src/test/java/org/openmrs/module/atomfeed/api/utils/AtomfeedUtilsTest.java [hunk 4]
```diff
@@ -9,49 +9,81 @@
...
+		expectedFeedConfiguration.setFeedWriter("custom.PatientWriter");
+		final HashMap<String, String> expectedLinkTemplates = new HashMap<>();
+		expectedLinkTemplates.put("rest", "openmrs/ws/rest/v1/patient{uuid}?v=full");
+		expectedLinkTemplates.put("fhir", "openmrs/ws/fhir/v1/patient{uuid}?v=full");
+		expectedFeedConfiguration.setLinkTemplates(expectedLinkTemplates);
+	}
 	
 	@Test
-	public void readResourceFile_shouldReadSampleFile() throws AtomfeedIoException {
+	public void readResourceFile_shouldReadSampleFile() throws AtomfeedException {
```
Comment (pkornowski) [inline api/src/test/java/org/openmrs/module/atomfeed/api/utils/AtomfeedUtilsTest.java]: Change name convention.

File: api/src/test/java/org/openmrs/module/atomfeed/api/utils/FeedConfigurationManagerTest.java [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+        FeedConfiguration[] array = AtomfeedUtils.parseJsonFileToFeedConfiguration(sampleFeedConfigurationPath);
+        FeedConfigurationManager manager = new FeedConfigurationManager();
+        manager.loadLocalFeedConfiguration(array);
+        Assert.assertEquals(expectedFeedConfiguration, manager.getFeedConfigurationByTitle("Title"));
+    }
+
+    @Test
+    public void loadLocalFeedConfiguration_shouldLoadFeedConfigurationFromStringCorrectly() throws AtomfeedException {
+        final HashMap<String, String> links = new HashMap<>();
+        links.put("rest", "openmrs/ws/rest/v1/patient{uuid}?v=full");
```
Comment (pkornowski) [inline api/src/test/java/org/openmrs/module/atomfeed/api/utils/FeedConfigurationManagerTest.java]: It's used twice. Please, move this string to the private static final.

Comment (areklalo) [reply api/src/test/java/org/openmrs/module/atomfeed/api/utils/FeedConfigurationManagerTest.java]: Corrected.

**Comment by pkornowski** [comment]
@areklalo 
Resolve conflicts, please.

**Comment by areklalo** [comment]
Work will be continued here: https://github.com/openmrs/openmrs-module-atomfeed/pull/6 because of some issues with the fork.

**Pull request closed without merging**